wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/PluginMetaRetriever.java

```diff
@@ -115,67 +117,89 @@ public PluginMetaRetriever(
       throw new InfrastructureException(CHE_REGISTRY_MISSING_ERROR);
     }
 
-    ArrayList<Pair<String, String>> metasIdsVersions = new ArrayList<>();
+    List<PluginFQN> metaFQNs = new ArrayList<>();
     if (!isNullOrEmpty(pluginsAttribute)) {
       String[] plugins = pluginsAttribute.split(" *, *");
       if (plugins.length != 0) {
-        Collection<Pair<String, String>> pluginsIdsVersions = parseIdsVersions(plugins);
-        metasIdsVersions.addAll(pluginsIdsVersions);
+        Collection<PluginFQN> pluginsFQNs = parsePluginFQNs(plugins);
+        metaFQNs.addAll(pluginsFQNs);
       }
     }
     if (!isNullOrEmpty(editorAttribute)) {
-      Collection<Pair<String, String>> editorIdVersionCollection =
-          parseIdsVersions(editorAttribute);
+      Collection<PluginFQN> editorIdVersionCollection =
+          parsePluginFQNs(editorAttribute.split(" *, *"));
       if (editorIdVersionCollection.size() > 1) {
         throw new InfrastructureException(
             "Multiple editors found in workspace config attributes. "
                 + "It is not supported. Please, use one editor only.");
       }
-      metasIdsVersions.addAll(editorIdVersionCollection);
+      metaFQNs.addAll(editorIdVersionCollection);
     }
 
-    return getMetas(metasIdsVersions);
+    return getMetas(metaFQNs);
   }
 
-  private Collection<Pair<String, String>> parseIdsVersions(String... idsVersions)
-      throws InfrastructureException {
-    Map<String, Pair<String, String>> collectedIdVersion = new HashMap<>();
-    for (String plugin : idsVersions) {
-      String[] idVersion = plugin.split(":");
+  private Collection<PluginFQN> parsePluginFQNs(String... plugins) throws InfrastructureException {
+    List<PluginFQN> collectedFQNs = new ArrayList<>();
+    for (String plugin : plugins) {
+      URI repo = null;
+      String idVersionString;
+      final int idVersionTagDelimiter = plugin.lastIndexOf("/");
+      idVersionString = plugin.substring(idVersionTagDelimiter + 1);
+      if (idVersionTagDelimiter > -1) {
+        try {
+          repo = new URI(plugin.substring(0, idVersionTagDelimiter));
+        } catch (URISyntaxException e) {
+          throw new InternalInfrastructureException(
```

Comment:
Since `plugin` value that is provided by a user, I don't think that`InternalInfrastructureException` should be thrown here

Reply:
+1